### PR TITLE
Clarify what should be sent on PTO

### LINKED ISSUE #3780: recovery: clarifying the priority of Probe packet contents
Currently "Sending Probe Packets" define the priority what to be in Probe packet as follows:

1. A probe packet SHOULD carry new data when possible.
2. A probe packet MAY carry retransmitted unacknowledged data when new data is unavailable.

And also says "Implementations MAY use alternative strategies for determining the content of probe packets, including sending new or retransmitted data based on the application's priorities."

When there is no new data (case 2), the implementation need to choose which one to retransmit when there is a multiple data loss in the sender. QUIC's Probe packet approach is a combined experience of TCP TLP and RTO and I believe Probe packet is more similar to TLP. RACK draft says (https://datatracker.ietf.org/doc/html/draft-ietf-tcpm-rack-08#section-7.3) "The probe data segment can either be new, previously unsent data, or a retransmission of previously sent data just below SND.NXT" so it's pretty clear which one to send among the lost data (newest one).

Is there any reason that it's up to the implementers when it need to choose which one to send from lost data? It's better to clarify which one to send among retransmitted unacknowledged data, from oldest or newest when there is multiple data packet lost.

I think it's better to send newest (unacked) one first if I follow the reasoning of TLP (new to the old), but there may be some reason why the current draft doesn't define it clearly.

https://github.com/cloudflare/quiche/pull/562#discussion_r441864077 has some background and my comment.

### PR DESCRIPTION
**Pull request by ianswett**
I think this is clearer, but maybe there's a larger change people would like.

Fixes #3780

### CODE REVIEW

File: draft-ietf-quic-recovery.md [hunk 1]
```diff
@@ -640,8 +640,8 @@ spaces.
 If the sender wants to elicit a faster acknowledgement on PTO, it can skip a
 packet number to eliminate the ack delay.
 
-When the PTO timer expires, and there is new or previously sent unacknowledged
-data, it MUST be sent. A probe packet SHOULD carry new data when possible.
+When the PTO timer expires, new or previously sent unacknowledged data MUST
+be sent if available. A probe packet SHOULD carry new data when possible.
 A probe packet MAY carry retransmitted unacknowledged data when new data is
 unavailable, when flow control does not permit new data to be sent, or to
 opportunistically reduce loss recovery delay.  Implementations MAY use
```
Comment (martinthomson) [inline draft-ietf-quic-recovery.md:647]:
```suggestion
When the PTO timer expires, an ack-eliciting packet MUST be sent.  An endpoint
SHOULD include new data in this packet.  Previously sent data MAY be sent if
no new data can be sent.  Implementations MAY use
```

Comment (junhochoi) [reply draft-ietf-quic-recovery.md:647]: Thanks. I think this is more concise. Additionally,

"When the PTO timer ..." -> L625 says "When a PTO timer ..."
"an ack-eliciting packet MUST be sent" -> is it a duplicate of L625?
"An endpoint SHOULD include new data in this packet" -> "An endpoint SHOULD include new data in this packet if available"

Comment (junhochoi) [reply draft-ietf-quic-recovery.md:647]: Also, L649 (below this line, I can't comment on the line) says "sending new or retransmitted" -> "sending new or previously-sent"

Comment (martinthomson) [reply draft-ietf-quic-recovery.md:647]: Yeah the duplication with earlier is bad.

```suggestion
An endpoint SHOULD include new data in probe packets.  Data that was sent
previously MAY be sent again no new data can be sent.  Implementations MAY use
```

I think that using previously-sent and retransmitted is OK.

**Pull request merged**